hail/src/main/resources/include/hail/Utils.h

```diff
@@ -129,4 +133,19 @@ class ArrayLoadImpl : public BaseArrayImpl<elem_required, elem_size, elem_align>
   }
 };
 
+struct HailFatalError: public std::exception {
+  private:
+    static constexpr int max_error_len = 512;
+    char error_msg[max_error_len];
+  public:
+    virtual const char * what() const throw() { return error_msg; }
+    virtual ~HailFatalError() throw() {};
+    explicit HailFatalError(const char * fmtstring, ...) : std::exception() {
```

Comment:
Not for this PR, but when we can handle dependencies more easily, we should convert string formatting in C++ to fmt: http://fmtlib.net/latest/index.html